packages/contracts/src/Game.sol

```diff
@@ -312,7 +322,8 @@ contract Game {
     function fetchGameData(uint256 gameID) external view returns(FetchedGameData memory) {
         GameData storage gdata = gameData[gameID];
         PlayerData[] memory pData = new PlayerData[](gdata.players.length);
-        for (uint8 i = 0; i < gdata.players.length; ++i)
+        uint8 gdataLength = uint8(gdata.players.length);
+        for (uint8 i = 0; i < gdataLength; ++i)
```

Comment:
Super curious to know whether this makes a difference actually!

It's another case where a moderately smart compiler should be able to figure it out (nothing in the loop has side effects other than writing `pData`, so `gdata.playerData` should never change, so it should be possible to figure out the length doesn't change).

Can you compare both versions to find out?

Reply:
Yes, I'll compare both versions to find out, thanks :)
My motive was to save gas while calling SLOADs on the storage array for the length multiple times throught the iteration.
Will find out the gas savings.